### PR TITLE
Consistent order/degree in variable order spaces [var-order-space-no-degree]

### PR DESCRIPTION
**Pull request by jakubcerveny**
This is a modification of #1817 to resolve the problem of `FiniteElementSpace::GetElementOrder` and `GetElementDegree` returning different values for RT spaces. 

In this version, the vast majority of the code works with a single meaning of "order", which is the polynomial degree. The order of elements in `FiniteElementSpace` is the actual degree, which should match the return value of `FiniteElement::GetOrder`. The methods `GetElementDegree` and `GetFaceDegree` are gone. `FiniteElementSpace::GetOrder` is no longer deprecated and returns the same thing as `GetElementOrder`.

The only deviation is limited to `RT_FiniteElementCollection`, which takes an "order" in the RT sense, e.g., 0 means linear polynomials. `GetOrder` in the `FiniteElementCollection` base class will however return 1, and the default order of elements in the space is also 1. 

Also added a consistency check inside `FiniteElementSpace::GetFE` which verifies that `FiniteElementCollection::GetOrder` returns the same thing as `FiniteElement::GetOrder` (in debug build only).

### CODE REVIEW

**Comment by v-dobrev** [comment]
Based on the description, this sounds good to me. Thanks, Jakub!

**Comment by tzanio** [comment]
We don't need a formal review for this, since it is going into #1817.

I suggest we just merge it once we have two approvals.

**Review by tzanio — APPROVED** [review]
Thank you for making this change @jakubcerveny. I think it is an improvement.

**Review by pazner — APPROVED** [review]
Looks good to me. I think it's less confusing than before.

**Pull request merged**
